src/go/rdctl/cmd/snapshotList.go

```diff
@@ -69,6 +69,17 @@ func jsonOutput(snapshots []snapshot.Snapshot) error {
 		if err != nil {
 			return err
 		}
+		// And we need to hide the ID field
+		snapshotWithoutID := map[string]string{}
+		err = json.Unmarshal(jsonBuffer, &snapshotWithoutID)
+		if err != nil {
+			return err
+		}
+		delete(snapshotWithoutID, "id")
+		jsonBuffer, err = json.Marshal(snapshotWithoutID)
+		if err != nil {
+			return err
+		}
```

Comment:
This logic seems complex: Marshal, Unmarshal, Marshal once more.

Maybe it would be simpler to change the tag on the `ID` field to include `,omitempty`, and then simply `aSnapshot.ID = ""` before the `Marshal` call?

Reply:
I assumed that this struct was also used to write the `metadata.json` file, which would need the id. But I didn't check; it just seemed like an obvious assumption.

I agree that hiding the `id` from the JSON is not strictly necessary; maybe we should just leave it in. I just don't want it displayed in the regular user output.

Reply:
I tried setting `aSnapshot.ID = ""` and that left `"id": ""` in the JSON.

The problem is that `snapshotList` doesn't own the definition of `snapshot.Snapshot`

After all that work I did with `reflect`, sloshing the JSON through a `map[]interface{}` took no time and
worked the first time, but I'll look at a more elegant solution.

As for hiding the ID field, there's nothing a user can do with it except go look at the underlying contents. We don't expose the ID for docker contexts, and it's the same problem

Reply:
> I tried setting `aSnapshot.ID = ""` and that left `"id": ""` in the JSON.

That means you did not modify the tag on the ID field:

```go
ID      string    `json:"id,omitempty"`
```

> The problem is that snapshotList doesn't own the definition of snapshot.Snapshot

That's why I suggested the `omitempty` modifier instead of removing it completely (which I think wouldn't work).

`omitempty` is almost always fine; on Unmarshaling you still end up with the same empty string value as if you had written the field to the JSON.

Reply:
I put the `omitempty` in the wrong place, between the double-quote and the tilde instead of to the left of the double-quote.

Reply:
> I assumed that this struct was also used to write the metadata.json file

Right you are.

I'm still not sure why we can't just include `ID` in the JSON output...?